audio/decode/ad_lavc.c

```diff
@@ -60,23 +60,23 @@ struct priv {
 #define OPT_BASE_STRUCT struct ad_lavc_params
 struct ad_lavc_params {
     float ac3drc;
-    int downmix;
+    bool downmix;
     int threads;
     char **avopts;
 };
 
 const struct m_sub_options ad_lavc_conf = {
     .opts = (const m_option_t[]) {
         {"ac3drc", OPT_FLOAT(ac3drc), M_RANGE(0, 6)},
-        {"downmix", OPT_FLAG(downmix)},
+        {"downmix", OPT_BOOL(downmix)},
         {"threads", OPT_INT(threads), M_RANGE(0, 16)},
         {"o", OPT_KEYVALUELIST(avopts)},
         {0}
     },
     .size = sizeof(struct ad_lavc_params),
     .defaults = &(const struct ad_lavc_params){
         .ac3drc = 0,
-        .downmix = 0,
+        .downmix = false,
```

Comment:
Last thing but I'm pretty sure explicitly setting defaults as `false` isn't needed. It should always be false if you don't specify anything (applies to other parts of the code as well of course).

Reply:
I noticed that some get explicitly initialized why for most it's implicit. I've simply left it the way it was.
There are also plenty of integers that get explicitly initialized to 0, so those could also be removed it that's desired.

Reply:
Since this is touching all the bool options anyway, I say go and ahead and remove all the explicit false initializations for consistency. Having a mix just confuses people; I'm sure I'm guilty of it somewhere. And yeah removing integers that are 0 by default sounds like a good idea to me. That can be a separate commit.

Reply:
I've left some integer to 0 initialization when it made semantic sense (e.g. things like `offset_x = 1, offset_y = 0`).